cmd/cluster/detachstuckvolume.go

```diff
@@ -0,0 +1,166 @@
+	detachstuckvolumeCmd.MarkFlagRequired("cluster-id")
+	return detachstuckvolumeCmd
+
+}
+
+func newdetachStuckVolumeOptions() *detachStuckVolumeOptions {
+	return &detachStuckVolumeOptions{}
+}
+
+func (d *detachStuckVolumeOptions) complete(cmd *cobra.Command, args []string) error {
+
+	err := utils.IsValidClusterKey(d.clusterID)
+	if err != nil {
+		return err
+	}
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return err
+	}
+	defer connection.Close()
+	cluster, err := utils.GetCluster(connection, d.clusterID)
+	if err != nil {
+		return err
+	}
+	d.cluster = cluster
+	d.clusterID = cluster.ID()
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return errors.New("this command is only available for AWS clusters")
+	}
+
+	return nil
+}
+
+func (d *detachStuckVolumeOptions) run() error {
+
+	_, _, clientset, err := getKubeConfigAndClient(d.clusterID)
+
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Kubernetes configuration and client for cluster with ID %s: %w", d.clusterID, err)
+	}
+
+	err = volIdRegion(clientset, "openshift-monitoring", "")
+	if err != nil {
+		return err
+	}
+
+	// If the volIdRegion found no pv is utilized by non running state pod. Which make global variable nil.
+	// Thus there's a panic exit. In order to prevent it we're using following logic to prevent panic exit.
+	if len(Region) == 0 && len(VolumeId) == 0 {
+		return fmt.Errorf("there's no pv utilized by non running state pod in cluster: %s\nNo action prequired", d.clusterID)
+	}
+
+	if len(Region) != 1 {
+		fmt.Printf("Got more than one region value: %v", len(Region))
+	}
+	fmt.Println(Region[0])
+
+	fmt.Println(VolumeId)
+	// aws ec2 detach-volume --volume-id $VOLUME_ID --region $REGION --force
+	// WiP - Need to convert above cmd to function once volIdRegion gets completed
+
+	return nil
+
+}
+
+// Following function gets the volumeID & region of pv for non running state pod & value into global variable
+func volIdRegion(clientset *kubernetes.Clientset, namespace, selector string) error {
+
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list pods in namespace '%s'", NameSpace)
+
+	}
+	pV, err := clientset.CoreV1().PersistentVolumes().List(context.TODO(), v1.ListOptions{})
+
+	if err != nil {
+		return fmt.Errorf("failed to list pv in namespace '%s'", NameSpace)
+
+	}
+
+	for _, pod := range pods.Items {
+		//if pod.Status.Phase != "Running" { // IMP: We only need to pass the following condition for non running state pods. Will un-comment once the testing is complete.
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim != nil {
+				for _, pVol := range pV.Items {
+					if pVol.Spec.AWSElasticBlockStore != nil {
+						// Most of the cluster return AWSElasticBlockStore as nil. Could write code, not sure what'll be actual response.
+						fmt.Println("Gathering info from AWSElastic")
+						// Logic -
```

Comment:
Is this the actual logic or something else needs to be done

Reply:
The jira ticket migration script detach AWSElasticBlockStore id's but as per Florian mentioned in thread conversion that go with cis volume most of the cluster doesn't have AWSElasticBlockStore. Hence i just left blank